File: 2020/02/b.js
const fs = require('fs');

console.log(fs.readFileSync('input', 'utf8').trim().split('\n')
	.map(l => l.split(/(\d+)-(\d+) (.): (.+)/))
	.map(l => [parseInt(l[1]), parseInt(l[2]), l[3], l[4]])
	.reduce((p, [p1, p2, ch, pwd]) => {
		const ok =
			(pwd[p1 - 1] === ch && pwd[p2 - 1] !== ch) ||
			(pwd[p1 - 1] !== ch && pwd[p2 - 1] === ch);
		return p + (ok ? 1 : 0);
}, 0));

File: 2020/02/b.go
package main

import (
	"io/ioutil"
	"log"
	"regexp"
	"strconv"
	"strings"
)

func main() {
	f, err := ioutil.ReadFile("input")
	if err != nil {
		log.Fatalf("%v\n", err)
	}
	data := strings.Split(strings.TrimSpace(string(f)), "\n")

	// 1-3 b: cdefg
	re := regexp.MustCompile(`(\d+)-(\d+) (.): (.+)`)

	valid := 0

	for _, line := range data {
		m := re.FindStringSubmatch(line)
		p1s,p2s, ch, pwd := m[1], m[2], m[3][0], m[4]
		p1, _ := strconv.Atoi(p1s)
		p2, _ := strconv.Atoi(p2s)
		if ((pwd[p1 - 1] == ch && pwd[p2 - 1] != ch) ||
			(pwd[p1 - 1] != ch && pwd[p2 - 1] == ch)) {
			valid++
		}
	}

	log.Print(valid)
}
